File: 487-max-consecutive-ones-ii/max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        
        max_w = 0
        num_zeros = 0
        n = len(nums)
        l = 0
        
        for r in range(n):
            if nums[r] == 0:
                num_zeros += 1
            
            while num_zeros > 1:
                if nums[l] == 0:
                    num_zeros -= 1
                l += 1
            
            w = r - l + 1
            max_w = max(max_w, w)
        
        return max_w

        

File: 487-max-consecutive-ones-ii/max-consecutive-ones-ii.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int ans=0;
        int count=0;
        int j=-1;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0){
                count++;
            }
            while(count>1){
                j++;
                if(nums[j]==0) count--;
            }
            int len=i-j;
            if(len>ans) ans=len;
        }
        return ans;
    }
}